### PR TITLE
Maeloni lesson backend

### PR DESCRIPTION
**Pull request by Mae830**
Changed the controller for the module so that the lessons will be deleted as their corresponding lesson is deleted. This should only have a change in the module controller.

### CODE REVIEW

**Comment by kapooramanpreet** [comment]
Hi Maeloni, Can you do a git pull from main into this branch and push the code again? I feel like there are a lot of files to review which should not be the case here.

**Comment by Mae830** [comment]
Hey yeah I can do that sorry for seeing this late

File: src/api/controllers/module.controller.ts [hunk 1]
```diff
@@ -236,6 +237,21 @@ export const deleteModule = async (
         return;
       }
 
+      //Delete the lesson arrays
+      try {
+        for (let i = 0; i < module.problems.length; i++) {
```
Comment (marcgabe15) [inline src/api/controllers/module.controller.ts:242]: module.lessons?

**Review by marcgabe15 — CHANGES_REQUESTED** [review]

**Comment by Mae830** [comment]
Done. Tests now pass thank you for finding this

**Review by kapooramanpreet — CHANGES_REQUESTED** [review]
I thiunk Lin should also be module.lessons[i]

**Review by kapooramanpreet — APPROVED** [review]
Thanks for addressing the changes.

**Review by digitdustin — APPROVED** [review]

**Comment by kapooramanpreet** [comment]
Added requested changes.

**Pull request merged**